### PR TITLE
RFC 50: Remove hardcoded paths and use path_helpers instead (in the specs)

### PR DESCRIPTION
**Pull request by Ridhwana**
This was a painfully laborious PR 😭 

## What type of PR is this? (check all applicable)

- [ ] Refactor
- [x] Feature
- [x] Bug Fix
- [x] Optimization
- [ ] Documentation Update

☝️ It's a combination of all of these, I guess 🤔

## Description

These are a series of PR's and so it's difficult to provide context but I'm going to try and summarise it: 

As part of restructuring our admin navigation, we've changed our routes around. Soon we will be removing the old routes (and the feature flag) in favour of the new routes that contain scopes. These can be found in https://github.com/forem/forem/blob/main/config/routes/admin.rb.  You can find more info about the RFC [here](https://github.com/forem/rfcs/pull/50)

Hence, I've done the following in this PR:
- I've removed all the hardcoded routes and replaced them with the rails path helper (which also points to the new routes once the old routes are removed)
- I've updated the RSpec description, and context blocks to contain the new routes. Right now its a bit of a mismatch because with the feature flag disabled the tests are pointing to the old routes, but [without making these descriptions dynamic](https://forem-team.slack.com/archives/C7GE84DEJ/p1619467183497000) as well, this is the best we can do. In any case we planning on wrapping this feature up this week so it won't be too long until the path helpers are also referencing the new routes. Hence, I think that this is acceptable. 

What is not covered in this PR?
- I haven't yet updated the cypress routes to point to the new routes since we can't use path helpers in this part of the code base, hence I'll change it when I delete all the old routes.

## Related Tickets & Documents
https://github.com/forem/rfcs/pull/50

## QA Instructions, Screenshots, Recordings
- Double check that mappings are correct from hardcoded routes to dynamic path helpers 
- Optional: Grep through the repository for any hardcoded /admin routes left behind that need to be changed to path helpers. These routes will not include the user related, permissions and invitation routes as they have not changed so I've not updated all of them to use path helpers as yet. You can reference [this file](https://github.com/forem/forem/blob/main/config/routes/admin.rb) for routes that would have needed to change. This will also exclude the nested sidebar. 

### UI accessibility concerns?

_If your PR includes UI changes, please replace this line with details on how
accessibility is impacted and tested. For more info, check out the
[Forem Accessibility Docs](https://docs.forem.com/frontend/accessibility)._

## Added tests?

- [ ] Yes
- [x] No, and this is why: I changed to path helpers so need to ensure that all tests path.
- [ ] I need help with writing tests

## [Forem core team only] How will this change be communicated?

_Will this PR introduce a change that impacts Forem members or creators, the
development process, or any of our internal teams? If so, please note how you
will share this change with the people who need to know about it._

- [ ] I've updated the [Developer Docs](https://docs.forem.com) and/or
      [Admin Guide](https://forem.gitbook.io/forem-admin-guide/), or
      [Storybook](https://storybook.forem.com/) (for Crayons components)
- [ ] I've updated the README or added inline documentation
- [ ] I will share this change in a [Changelog](https://forem.dev/t/changelog)
      or in a [forem.dev](http://forem.dev) post
- [ ] I will share this change internally with the appropriate teams
- [ ] I'm not sure how best to communicate this change and need help
- [x] This change does not need to be communicated, and this is why not: _please
      replace this line with details on why this change doesn't need to be
      shared_

## [optional] What gif best describes this PR or how it makes you feel?

alt_text

### CODE REVIEW

File: spec/requests/admin/badge_achievements_spec.rb [hunk 1]
```diff
@@ -1,7 +1,7 @@
 require "rails_helper"
 require "requests/shared_examples/internal_policy_dependant_request"
 
-RSpec.describe "/admin/badges", type: :request do
```
Comment (Ridhwana) [inline spec/requests/admin/badge_achievements_spec.rb:4]: Seemed like this file is actually testing badge_achievements and not badges

File: spec/requests/admin/chat_channel_spec.rb [hunk 2]
```diff
@@ -20,38 +20,38 @@
...
+      patch admin_chat_channel_path(chat_channel.id),
             params: { chat_channel: { usernames_string: second_user.username } },
             headers: { HTTP_ACCEPT: "application/json" }
       expect(second_user.chat_channel_memberships.last).not_to be_blank
       expect(second_user.chat_channel_memberships.last.role).to eq "member"
     end
   end
 
-  describe "DELETE /admin/chat_channels/remove_user/:id" do
+  describe "DELETE /admin/apps/chat_channels/:id/remove_user" do
```
Comment (Ridhwana) [inline spec/requests/admin/chat_channel_spec.rb:36]: This was the wrong way around and so I changed the URL to be correct

File: spec/requests/admin/consumer_apps_spec.rb [hunk 3]
```diff
@@ -2,15 +2,15 @@
 require "requests/shared_examples/internal_policy_dependant_request"
 
 RSpec.describe "/admin/consumer_apps", type: :request do
```
Comment (Ridhwana) [inline spec/requests/admin/consumer_apps_spec.rb:4]: These routes have not changed and so we've kept the descriptions the same

File: spec/requests/admin/feedback_messages_spec.rb [hunk 4]
```diff
@@ -28,7 +28,7 @@
     end
   end
 
-  describe "POST /save_status" do
```
Comment (Ridhwana) [inline spec/requests/admin/feedback_messages_spec.rb:31]: made the descriptions here more verbosee

File: spec/requests/admin/invitations_spec.rb [hunk 5]
```diff
@@ -11,28 +11,28 @@
   describe "GET /admin/invitations" do
     it "renders to appropriate page" do
       user.update_column(:registered, false)
-      get "/admin/invitations"
```
Comment (Ridhwana) [inline spec/requests/admin/invitations_spec.rb:14]: These paths stayed the same with the admin navigation restructure but I just changed them to use path helpers anyways

File: spec/requests/admin/moderator_logs_spec.rb [hunk 6]
```diff
@@ -15,7 +15,7 @@
     Audit::Subscribe.forget listener
   end
 
-  describe "PUT /admin/tag/:id" do
```
Comment (Ridhwana) [inline spec/requests/admin/moderator_logs_spec.rb:18]: changed the description to the correct path

File: spec/requests/admin/users_manage_spec.rb [hunk 7]
```diff
@@ -68,7 +68,7 @@ def full_profile
     end
 
     it "deletes duplicate user" do
-      post "/admin/users/#{user.id}/merge", params: { user: { merge_user_id: user2.id } }
```
Comment (Ridhwana) [inline spec/requests/admin/users_manage_spec.rb:71]: These paths also stay the same with the restructure, but whilst I was here I just moved them over to path helpers

File: spec/requests/admin/users_spec.rb [hunk 8]
```diff
@@ -1,6 +1,6 @@
 require "rails_helper"
 
-RSpec.describe "admin/users", type: :request do
```
Comment (Ridhwana) [inline spec/requests/admin/users_spec.rb:3]: same with these, not impacted by the restructure

**Review by rhymes — APPROVED** [review]
Thanks for taking this on. That's A LOT of hardcoded paths. Definitely something to mention to the team for future PRs :D 

I found a few more for Part 2:

```text
spec/requests/user/user_profile_spec.rb:228:      get "/#{user.username}/admin"
spec/requests/admin/permissions_spec.rb # lines 13, 23, 36 and 53
spec/requests/admin/invitations_spec.rb:54
spec/requests/articles/articles_admin_unpublish_spec.rb:14
```

File: spec/requests/admin/articles_spec.rb [hunk 9]
```diff
@@ -1,12 +1,12 @@
 require "rails_helper"
 require "requests/shared_examples/internal_policy_dependant_request"
 
-RSpec.describe "/admin/articles", type: :request do
+RSpec.describe "/admin/content_manager/articles", type: :request do
```
[the remark this answers is not in the document]
Comment (rhymes) [reply spec/requests/admin/articles_spec.rb:4]: I would have just wrote `describe "Admin Articles"` 🤣

[the remark this answers is not in the document]
Comment (Ridhwana) [reply spec/requests/admin/articles_spec.rb:4]: oh my gosh, no Rhymes, why do you have to come up with such good ideas! I should have done that!

**Comment by Ridhwana** [comment]
> spec/requests/user/user_profile_spec.rb:228:      get "/#{user.username}/admin"
> spec/requests/admin/permissions_spec.rb # lines 13, 23, 36 and 53
> spec/requests/admin/invitations_spec.rb:54
> spec/requests/articles/articles_admin_unpublish_spec.rb:14

Thanks for these Rhymes, I'll add them on a part two. None of these affect the admin interface at the moment because they're routes that haven't changed 😊. 

Also I don't think these (`spec/requests/user/user_profile_spec.rb:228 get "/#{user.username}/admin"
` and `spec/requests/articles/articles_admin_unpublish_spec.rb:14`) would be converted since it's not under`/admin`

**Review by juliannatetreault — APPROVED** [review]
Thank you so much for taking the time to do this!! ✨

**Pull request merged**